mir-webapp/src/main/solr/docker-entrypoint.sh

```diff
@@ -0,0 +1,108 @@
+
+function fixDirectoryRights() {
+  find "$1" \! -user "$2" -exec chown "$2:$2" '{}' +
+}
+
+echo "Running solr entry script as user: $(whoami)"
+if [ "$(id -u)" -eq 0 ]; then
+  fixDirectoryRights /var/solr/ solr
+  exec /usr/bin/sudo -E -u solr "PATH=$PATH" "$(pwd)/$0";
+  exit 0;
+fi
+
+function solrpass() {
+  printf "%s %s" "$(echo -n "$2$1"|openssl dgst -sha256 -binary|openssl dgst -sha256 -binary|openssl base64)" "$(echo -n "$2"|openssl base64)"
+}
+
+secruity_json=/var/solr/data/security.json;
+
+
+echo "{" > $secruity_json
+echo "  \"authentication\":{" >> $secruity_json
+echo "    \"blockUnknown\": true," >> $secruity_json
+echo "    \"class\":\"solr.BasicAuthPlugin\"," >> $secruity_json
+echo "    \"credentials\": {" >> $secruity_json
+
+if [ -n "$SOLR_SEARCH_USER" ]; then
+  echo "      \"${SOLR_SEARCH_USER}\":\"$(solrpass $SOLR_SEARCH_PASSWORD $(openssl rand 10))\"," >> $secruity_json
+fi
+
+if [ -n "$SOLR_INDEX_USER" ]; then
+  echo "      \"${SOLR_INDEX_USER}\":\"$(solrpass $SOLR_INDEX_PASSWORD $(openssl rand 10))\"," >> $secruity_json
+fi
+
+if [ -n "$SOLR_ADMIN_USER" ]; then
+  echo "      \"${SOLR_ADMIN_USER}\":\"$(solrpass $SOLR_ADMIN_PASSWORD $(openssl rand 10))\"" >> $secruity_json
+fi
+
+echo "    }," >> $secruity_json
+echo "    \"realm\":\"My Solr users\"," >> $secruity_json
+echo "    \"forwardCredentials\": false" >> $secruity_json
+echo "  }," >> $secruity_json
+echo "  \"authorization\":{" >> $secruity_json
+echo "    \"class\":\"solr.RuleBasedAuthorizationPlugin\"," >> $secruity_json
+echo "    \"permissions\":[" >> $secruity_json
+
+if [ -n "$SOLR_SEARCH_USER" ]; then
+  echo "      {" >> $secruity_json
+  echo "        \"name\":\"read\"," >> $secruity_json
+  echo "        \"role\":[\"searcher\",\"indexer\",\"admin\"]" >> $secruity_json
+  echo "      }," >> $secruity_json
+fi
+
+if [ -n "$SOLR_INDEX_USER" ]; then
+  echo "      {" >> $secruity_json
+  echo "        \"name\":\"update\"," >> $secruity_json
+  echo "        \"role\":[\"indexer\",\"admin\"]" >> $secruity_json
+  echo "      }," >> $secruity_json
+fi
+
+if [ -n "$SOLR_ADMIN_USER" ]; then
+  echo "      {" >> $secruity_json
+  echo "        \"name\":\"all\"," >> $secruity_json
+  echo "        \"role\":\"admin\"" >> $secruity_json
+  echo "      }" >> $secruity_json
+fi
+echo "]," >> $secruity_json
+echo "    \"user-role\":{" >> $secruity_json
+
+if [ -n "$SOLR_SEARCH_USER" ]; then
+  echo "      \"${SOLR_SEARCH_USER}\":\"searcher\"," >> $secruity_json
+fi
+
+if [ -n "$SOLR_INDEX_USER" ]; then
+  echo "      \"${SOLR_INDEX_USER}\":\"indexer\"," >> $secruity_json
+fi
+
+if [ -n "$SOLR_ADMIN_USER" ]; then
+  echo "      \"${SOLR_ADMIN_USER}\":\"admin\"" >> $secruity_json
+fi
+
+echo "    }" >> $secruity_json
+echo "  }" >> $secruity_json
+echo "}" >> $secruity_json
+
+(/opt/docker-solr/scripts/wait-for-solr.sh;/opt/solr/bin/solr zk cp $secruity_json zk:security.json -z localhost:9983)&
+/opt/docker-solr/scripts/solr-foreground -c;
```

Comment:
please check that EVERY files ends with an empty newline

Reply:
should be fine now